include/nbl/asset/ICPUBufferView.h

```diff
@@ -96,6 +105,12 @@ class ICPUBufferView : public IBufferView<ICPUBuffer>, public IAsset
 		}
 
 		virtual ~ICPUBufferView() = default;
+
+	private:
+
+		bool compareMembers(const ICPUBufferView* other) const {
+			return (m_size == other->m_size) && (m_offset == other->m_offset) && (m_format == other->m_format);
+		}
```

Comment:
ok we want to move it up to `IAsset` and call it something like `protected: bool compatible`

then every asset's `canBeRestoredFrom` and `equals` can call into `compatible()` first